frontend/src/core/comments/components/CommentsList.tsx

```diff
@@ -67,13 +67,13 @@ export default function CommentsList(props: Props): React.ReactElement<'div'> {
                             renderComment(comment),
                         )}
                     </ul>
-
-                    <Localized id='comments-CommentsList--all-comments'>
-                        <h2 className='title'>ALL COMMENTS</h2>
-                    </Localized>
                 </section>
             ) : null}
             <section className='all-comments'>
+                <Localized id='comments-CommentsList--all-comments'>
+                    <h2 className='title'>ALL COMMENTS</h2>
+                </Localized>
```

Comment:
This will render the All Comments title even when we don't have any Pinned comments, which is the case most of the time. I don't think we need this change, so I'd revert.

Reply:
I'm a bit confused on this, 

1. when the user is signed in, we want the All Comments title and section where they can comment regardless of whether there are Pinned comments or not?

2. when there is no user signed in, we want the All Comment title and section to be hidden when there are only Pinned Comments. When there are no pinned comments and no comments, All Comment titles and sections are hidden?

Reply:
1. Only the section (because `<AddComment>`is within it). We only add titles when we have Pinned comments.

2. Yes and yes.

Everything needs to be stay the same as it was in your original patch, except what I wrote in my comment above: "Don't render the All Comments title and All Comments panel if the user cannot add a comment and we only have pinned comments.".